### PR TITLE
Skip 0 byte pseudo kernel processes

### PR DESCRIPTION
**Pull request by rwaffen**
Fixes: #105

### CODE REVIEW

**Review by djberg96 — CHANGES_REQUESTED** [review]
I'd rather not skip everything if we don't have to if we can at least collect *some* information for the process. Personally I think we should just add `Errno::EBADF` to the list of rescued exceptions on line 346.

**Comment by djberg96** [comment]
Extra bonus points if you can integrate https://github.com/vmactions/solaris-vm :)

**Comment by djberg96** [comment]
Thanks for the update and taking a stab at the github action. However, it looks like it failed. I see `The process '/bin/bash' failed with exit code 1`.

File: .github/workflows/ruby.yml [hunk 1]
```diff
@@ -40,3 +40,19 @@ jobs:
...
+    steps:
+    - uses: actions/checkout@v2
+    - name: Test in Solaris
+      id: test
+      uses: vmactions/solaris-vm@v0
+      with:
+        usesh: true
+        prepare: |
+          pkgutil -y -i ruby
+
```
Comment (djberg96) [inline .github/workflows/ruby.yml:56]: I'm thinking we need to add a 'gem install bundler` here, followed by a `bundle install`. What do you think?

File: .github/workflows/ruby.yml [hunk 2]
```diff
@@ -40,3 +40,19 @@ jobs:
...
+        run: |
+          ruby -v
```
Comment (djberg96) [inline .github/workflows/ruby.yml:58]: I think we'll want `bundle exec rake` here.

File: .github/workflows/ruby.yml [hunk 3]
```diff
@@ -40,3 +40,19 @@ jobs:
         bundler-cache: true # runs 'bundle install' and caches installed gems automatically
     - name: Run tests
       run: bundle exec rake
+
+  test_solaris:
+    runs-on: macos-12
+    name: A job to run test in Solaris
```
Comment (djberg96) [inline .github/workflows/ruby.yml:46]: Let's just called is "test Solaris". :)

**Comment by djberg96** [comment]
I don't think we need the `ruby -v` command do we?

Beyond that, looking at the details I see `Package runtime/ruby-26 not in catalog. Exiting.`

It's up to you if you would like to continue to pursue this, I certainly won't make it a requirement for the main fix. I just thought it would be nice to have, especially since I could then crib from it for other projects. :)

**Comment by rwaffen** [comment]
ruby -v is only for testing which version would be installed, if it would install at all 🤦 
but i would say we split this, first fixing the issue and than separately trying to get this GHA working.
will remove the yaml or comment it out for the moment, so we than can continue later on.

**Comment by djberg96** [comment]
Ok, go ahead and remove it (or comment it out) and I'll re-run the jobs (which I understand won't actually be testing this change), then I'll merge it. :)

**Review by djberg96 — APPROVED** [review]

**Pull request merged**